File: AIdemosite/speech_translate.py
import azure.cognitiveservices.speech as speechsdk

def translate_from_speech(language):
    speech_translation_config = speechsdk.translation.SpeechTranslationConfig(subscription="ac1b8fe3034344a787dc45b06243f51b", region="eastasia")
    speech_translation_config.speech_recognition_language=language

    target_languages = ['zh-Hant', 'en', 'es', 'fr', 'de', 'ja']

    for lan in target_languages:
        speech_translation_config.add_target_language(lan)

    audio_config = speechsdk.audio.AudioConfig(use_default_microphone=True)
    translation_recognizer = speechsdk.translation.TranslationRecognizer(translation_config=speech_translation_config, audio_config=audio_config)

    print("Speak into your microphone.")
    translation_recognition_result = translation_recognizer.recognize_once_async().get()

    if translation_recognition_result.reason == speechsdk.ResultReason.TranslatedSpeech:
        translate_result = {}
        for lan in target_languages:
            translate_result[lan] = translation_recognition_result.translations[lan]
            # print("Translated into '{}':{}".format(lan, translation_recognition_result.translations[lan]))
        return translate_result
    elif translation_recognition_result.reason == speechsdk.ResultReason.NoMatch:
        for lan in target_languages:
            translate_result[lan] = "No speech could be recognized"
        print("No speech could be recognized: {}".format(translation_recognition_result.no_match_details))
        return translate_result
    elif translation_recognition_result.reason == speechsdk.ResultReason.Canceled:
        cancellation_details = translation_recognition_result.cancellation_details
        print("Speech Recognition canceled: {}".format(cancellation_details.reason))
        if cancellation_details.reason == speechsdk.CancellationReason.Error:
            print("Error details: {}".format(cancellation_details.error_details))
            print("Did you set the speech resource key and region values?")

# print(translate_from_speach("en-US"))

File: static/js/speech_translate.js
// status fields and start button in UI
var resultsDivs;
var btn_translate;
var subscriptionKey = "ac1b8fe3034344a787dc45b06243f51b",
  serviceRegion = "eastasia";
// subscription key and region for speech services.
var languageTargetOptions;
var SpeechSDK;
var recognizer;

document.addEventListener("DOMContentLoaded", function () {
  btn_translate = document.getElementById("btn_translate");
  lang_slct = document.getElementById("lang_slct");
  languageTargetOptions = ["zh-Hant", "en", "es", "fr", "de", "ja"];
  resultsDivs = [
    document.getElementById("output_zh"),
    document.getElementById("output_en"),
    document.getElementById("output_es"),
    document.getElementById("output_fr"),
    document.getElementById("output_de"),
    document.getElementById("output_ja"),
  ];

  btn_translate.addEventListener("click", function () {
    btn_translate.disabled = true;
    resultsDivs.forEach(function (elem) {
      elem.value = "";
    });

    if (subscriptionKey === "" || subscriptionKey === "subscription") {
      alert(
        "Please enter your Microsoft Cognitive Services Speech subscription key!"
      );
      btn_translate.disabled = false;
      return;
    }
    var speechConfig = SpeechSDK.SpeechTranslationConfig.fromSubscription(
      subscriptionKey,
      serviceRegion
    );
    console.log(lang_slct.value);
    speechConfig.speechRecognitionLanguage = lang_slct.value;
    let languageKeys = {};
    languageTargetOptions.forEach(function (langElem, index) {
      let language = langElem;
      languageKeys[language] = resultsDivs[index];
      speechConfig.addTargetLanguage(language);
    });

    var audioConfig = SpeechSDK.AudioConfig.fromDefaultMicrophoneInput();
    recognizer = new SpeechSDK.TranslationRecognizer(speechConfig, audioConfig);

    recognizer.recognizeOnceAsync(
      function (result) {
        btn_translate.disabled = false;
        if (result.reason === SpeechSDK.ResultReason.TranslatedSpeech) {
          for (var key in languageKeys) {
            let translation = result.translations.get(key);
            window.console.log(key + ": " + translation);
            languageKeys[key].value += translation;
          }
        }

        recognizer.close();
        recognizer = undefined;
      },
      function (err) {
        btn_translate.disabled = false;
        resultsDiv[0].innerHTML += err;
        window.console.log(err);

        recognizer.close();
        recognizer = undefined;
      }
    );
  });

  if (!!window.SpeechSDK) {
    SpeechSDK = window.SpeechSDK;
    btn_translate.disabled = false;

    // document.getElementById("content").style.display = "block";
    // document.getElementById("warning").style.display = "none";
  }
});
